lib/phoenix_component.ex

```diff
@@ -1630,15 +1601,26 @@ defmodule Phoenix.Component do
       </.form>
   """
   @doc type: :component
-  attr.(:for, :any, required: true)
-  attr.(:action, :string, default: nil)
-  attr.(:as, :atom, [])
-  attr.(:multipart, :boolean, default: false)
-  attr.(:method, :string, [])
-  attr.(:csrf_token, :boolean, [])
-  attr.(:errors, :list, [])
-  attr.(:rest, :global, [])
-  slot.(:inner_block, required: true)
+  attr.(:for, :any, required: true, doc: "The form source data.")
+  attr.(:action, :string, doc: "The action to submit the form on.
+  This attribute must be given if you intend to submit the form to a URL without LiveView.")
+  attr.(:as, :atom, doc: "The server side parameter in which all params for this form 
+  will be collected (i.e. `as: :user_params` would mean all fields for this form will be 
+  accessed as `conn.params.user_params` server side).
+  Automatically inflected when a changeset is given.")
+  attr.(:multipart, :boolean, default: false, doc: "Sets `enctype` to `multipart/form-data`.
+  Required when uploading files.")
+  attr.(:method, :string, doc: "The HTTP method.
+  It is only used if an `:action` is given. If the method is not `get` nor `post`,
+  an input tag with name `_method` is generated alongside the form tag.")
+  attr.(:csrf_token, :boolean, default: false, doc: "A token to authenticate the validity of requests. 
+  Once is automatically generated when an action is given and the method is not `get`. 
+  When set to `false`, no token is generated.")
+  attr.(:errors, :list, doc: "Use this to manually pass a keyword list of errors to the form, 
+  e.g. `conn.assigns[:errors]`. This option is only used when a connection is used as the form 
+  source and it will make the errors available under `f.errors`.")
```

Comment:
Can you please use heredocs for multilines? Otherwise the additional newlines and indentation will actually show up in the docs too. :)